File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/carts/domain/usecase/GetCartUseCaseTest.java
package com.tarbadev.witchcraft.carts.domain.usecase;

import com.tarbadev.witchcraft.carts.domain.usecase.GetCartUseCase;
import com.tarbadev.witchcraft.carts.domain.entity.Cart;
import com.tarbadev.witchcraft.carts.domain.repository.CartRepository;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;
import org.springframework.test.context.ActiveProfiles;

import static org.assertj.core.api.Java6Assertions.assertThat;
import static org.mockito.BDDMockito.given;

@RunWith(MockitoJUnitRunner.class)
@ActiveProfiles("test")
public class GetCartUseCaseTest {
  @Mock private CartRepository cartRepository;

  private GetCartUseCase subject;

  @Before
  public void setUp() {
    subject = new GetCartUseCase(cartRepository);
  }

  @Test
  public void execute() {
    Cart cart = Cart.builder().id(123).build();

    given(cartRepository.findById(123)).willReturn(cart);

    assertThat(subject.execute(123)).isEqualTo(cart);
  }
}

File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/carts/domain/usecase/GetCartUseCaseTest.kt
package com.tarbadev.witchcraft.carts.domain.usecase

import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.reset
import com.tarbadev.witchcraft.carts.domain.entity.Cart
import com.tarbadev.witchcraft.carts.domain.repository.CartRepository
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import com.nhaarman.mockitokotlin2.whenever

class GetCartUseCaseTest {
    private val cartRepository: CartRepository = mock()
    private val getCartUseCase: GetCartUseCase = GetCartUseCase(cartRepository)

    @BeforeEach
    fun setup() {
        reset(cartRepository)
    }

    @Test
    fun execute() {
        val cart = Cart(id = 123)

        whenever(cartRepository.findById(123)).thenReturn(cart)

        assertEquals(cart, getCartUseCase.execute(123))
    }
}